### DISCUSSION TITLE
[Support]: Home Assistant Events not honoring required_zones setting in 0.13

### DISCUSSION THREAD

**Discussion by TarheelGrad1998** [blakeblackshear/frigate#10298] [General Support]
### Describe the problem you are having

I upgraded to 0.13 today from 0.12, and some of my notifications in Home Assistant stopped triggering.  I have narrowed it down to the automation conditions with payload = "new" which checks for entered_zones.  On all of my "new" events, the entered_zones is empty.  This is despite the fact that I have required_zones specified for the cameras in my config.  Additionally, I see a ton of events sent to HA which are detecting vehicles not in any zone, resulting in excess noise.  This all worked perfectly in 0.12, so this functionality has changed with 0.13.

### Version

0.13.2-6476F8A

### Frigate config file

```yaml
cameras:
  front_doorbell:
...
    record: 
      enabled: True     
      events:
        required_zones:
          - driveway
          - yard
```

### Relevant log output

```shell
n/a
```

### FFprobe output from your camera

```shell
n/a
```

### Frigate stats

```json
n/a
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

n/a

### Any other information that may be helpful

    after:
      id: 1709766517.571947-vrdmer
      camera: front_doorbell
      frame_time: 1709766540.632798
      snapshot:
        frame_time: 1709766540.632798
        box:
          - 421
          - 219
          - 442
          - 229
        area: 210
        region:
          - 146
          - 71
          - 466
          - 391
        score: 0.7265625
        attributes: []
      label: vehicle
      sub_label: null
      top_score: 0.701171875
      false_positive: false
      start_time: 1709766517.571947
      end_time: null
      score: 0.7265625
      box:
        - 421
        - 219
        - 442
        - 229
      area: 210
      ratio: 2.1
      region:
        - 146
        - 71
        - 466
        - 391
      stationary: true
      motionless_count: 104
      position_changes: 1
      current_zones: []
      entered_zones: []
      has_clip: false
      has_snapshot: false
      attributes: {}
      current_attributes: []
    type: new

**Comment by NickM-27** [comment]
required_zones does not change what is sent in /events. /events has always been a raw topic that sends all information about an object and it is the job of the user to filter these items out, like using the entered_zones field for example.

Also, the functionality of sending zones for a `new` event has not changed either. It was never a guarantee that `new` would include a zone and I would not recommend triggering off of `new`, `update`, or `end` but rather triggering off of the information in the payload.

The reason you are seeing different behavior is because zones now have inertia so by default an object must be detected in a zone for multiple frames before it is considered in the zone in order to avoid misfires where an objects bounding box was inaccurate for a frame or two.

https://docs.frigate.video/configuration/zones#zone-inertia

**Comment by TarheelGrad1998** [comment]
I would argue, it is not an event if it does not meet the criteria I have set for an event.  It should not be sending events that by definition I have said are not events, because the object has never been in a required zone.  Thus, a new event in those cases should always contain a zone.

I can probably fix my automation by changing the conditions.  That doesn't change the excess noise, like how I'm getting events all day for my neighbor's cars that have not moved nor been in any of my zones.  

It behaved the way I described in 0.12.

**Reply by NickM-27** [reply]
that's not really correct. As the user you don't decide what is an event, you decide what events should have a snapshot / be recorded. 

The /events topic won't be changing behavior. Many users use its advanced information and I for example have notifications that for events that don't get recorded / snapshot.

There will be a new topic coming in 0.14 that will be a simplified version of /events that will be less verbose and work in the way that you are describing.

**Comment by TarheelGrad1998** [comment]
Thanks.  I'll deal with it in the automation, but I'm glad to hear an enhancement is in the works.  Having to define this event logic in two different places two different ways rankles me as a dev.

**Comment by TarheelGrad1998** [comment]
Also in case it helps anyone else, this condition (I think) restores the functionality of my notifications in HA:

{{ trigger.payload_json["type"] == "new" or not set(trigger.payload_json['after']['entered_zones']).issubset(set(trigger.payload_json['before']['entered_zones'])) }}

..in English, it's a new event, or the object has been detected in a new zone.
